lib/src/realm_object.dart

```diff
@@ -175,21 +186,23 @@ extension RealmObjectInternal on RealmObject {
     _accessor = accessor;
   }
 
-  static T create<T extends RealmObject>(RealmObjectHandle handle, RealmCoreAccessor accessor) {
-    if (!RealmObject._factories.containsKey(T)) {
-      throw Exception("Factory for object type $T not found.");
+  static RealmObject create(Type type, Realm realm, RealmObjectHandle handle, RealmCoreAccessor accessor) {
+    if (!RealmObject._factories.containsKey(type)) {
+      throw Exception("Factory for object type $type not found.");
     }
 
-    final object = RealmObject._factories[T]!() as T;
+    final object = RealmObject._factories[type]!();
```

Comment:
Maybe. this comment should go on another of the PRs, but consider making the factories take handle, accessor and realm as arguments, instead of setting them after construction.